recipe_scrapers/allrecipes.py

```diff
@@ -6,6 +6,19 @@ class AllRecipes(AbstractScraper):
     def host(cls):
         return "allrecipes.com"
 
+    def author(self):
+        # NB: In the schema.org 'Recipe' type, the 'author' property is a
+        # single-value type, not an ItemList.
+        # allrecipes.com seems to render the author property as a list
+        # containing a single item under some circumstances.
+        # In those cases, the SchemaOrg class will fail due to the unexpected
+        # type, and this method is called as a fallback.
+        # Rather than implement non-standard handling in SchemaOrg, this code
+        # provides a (hopefully temporary!) allrecipes-specific workaround.
+        author = self.schema.data.get("author")
+        if author and type(author) == list and len(author) == 1:
```

Comment:
Doesn't matter much for this scenario, but I find it best practice to use `isinstance` as opposed to `type() ==`.

https://stackoverflow.com/a/1549854

Reply:
Thanks, makes sense!  Fixup applied.